### PR TITLE
Doc update: Troubleshooting + SD libs

### LINKED ISSUE #1497: Failed to connect to ESP32: Timed out waiting for packet header
### Hardware:
Board:							ESP32 LOLIN w/OLED - ESP-WROM32
Core Installation/update date:		        (see below)
IDE name:						Arduino IDE 1.8.5
Flash Frequency:					80Mhz
Upload Speed:						Tried various

### Description:
Unable to upload anything to this module. 

Have another identical one and same sketch uploads and runs fine. Rules out cable, connection or computer. I tried to upload using esptool.py directly and also using FTDI but same problem.

How can I test if the board damaged? Any tips on how to revive it?

### Debug Messages:
(/dev/ttyUSB1)
ets Jun  8 2016 00:22:57

rst:0x10 (RTCWDT_RTC_RESET),boot:0x3 (DOWNLOAD_BOOT(UART0/UART1/SDIO_REI_REO_V2))
waiting for download
```
Sketch uses 638418 bytes (48%) of program storage space. Maximum is 1310720 bytes.
Global variables use 38948 bytes (13%) of dynamic memory, leaving 255964 bytes for local variables. Maximum is 294912 bytes.
esptool.py v2.3.1
Connecting........_____....._____....._____....._____....._____....._____....._____....._____....._____....._____

A fatal error occurred: Failed to connect to ESP32: Timed out waiting for packet header
A fatal error occurred: Failed to connect to ESP32: Timed out waiting for packet header
```
Thanks

### PR DESCRIPTION
**Pull request by PilnyTomas**
Troubleshooting:
- Updated flashing problems; closes #1497 
- Added paragraph describing S3 PSRAM options. Related discussion [#6426](https://github.com/espressif/arduino-esp32/discussions/6426), related issues #6980, ...  Related [article](https://randomnerdtutorials.com/solved-failed-to-connect-to-esp32-timed-out-waiting-for-packet-header/)
- Moved SPIFFS from FAQ to Troubleshooting
- SD card mount fail - closing #259 closing #7707; related: #8076 #963

SD libraries updates:
SD_MMC_Test.ino - added hint to read README
SD_MMC/README.md - new file, partially based on [IDF example](https://github.com/espressif/esp-idf/blob/903af13e847cd301e476d8b16b4ee1c21b30b5c6/examples/storage/sd_card/sdmmc/README.md#L4)
SD/README.md - extended

Minor updates:

- cdc_dfu_flash.rst - removed extra "it"
- io_mux.rst - added note about pins used for SPI flash
- SD_MMC.cpp  - typo fix

### CODE REVIEW

File: libraries/SD_MMC/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+GPIO12 is used as a bootstrapping pin to select output voltage of an internal regulator which powers the flash chip (VDD_SDIO). This pin has an internal pulldown so if left unconnected it will read low at reset (selecting default 3.3V operation). When adding a pullup to this pin for SD card operation, consider the following:
+
+## FAQ:
+
+#### Do I need any additional modules, like the Arduino SD module?
+
+No, just wire your SD card directly to ESP32.
+
+Tip: If you are using a microSD card and have a spare adapter to full-sized SD, you can solder Dupont pins on the adapter.
```
Comment (pedrominatel) [inline libraries/SD_MMC/README.md:99]: Dupont -> wires

File: libraries/SD_MMC/README.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+
+#### What is the difference between SD and SD_MMC libraries?
+
+SD runs on SPI, and SD_MMC uses the SDMMC hardware bus on the ESP32.
+The SPI uses 4 communication pins + 2 power connections and operates on up to 80MHz. The SPI option offers flexibility on pin connection because the data connections can be routed through GPIO matrix to any data pin.
+SD-SPI speed is approximately half of the SD-MMC even when used on 1-bit line.
+You can read more about SD SPI in the [documentation](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/sdspi_host.html)
+
+SD_MMC is supported only by ESP32 and ESP320S3 and can be connected only to dedicated pins. SD_MMC allows to use of 1, 4 or 8 data pins + 2 additional communication pins and 2 power pins. The data pins need to be pulled up externally.
```
Comment (pedrominatel) [inline libraries/SD_MMC/README.md:109]: ESP320S3 -> ESP32-S3

**Review by pedrominatel — COMMENTED** [review]
PTAL @PilnyTomas

File: libraries/SD_MMC/README.md (hunk 1 above)
Comment (PilnyTomas) [reply libraries/SD_MMC/README.md:99]: IMG_20230615_144850_926

Comment (pedrominatel) [reply libraries/SD_MMC/README.md:99]: This is called a 'pin header'.

Comment (pedrominatel) [reply libraries/SD_MMC/README.md:99]: To be more accurate, '2.54mm pin header'.

File: docs/source/troubleshooting.rst [hunk 3]
```diff
@@ -54,23 +54,27 @@ Possible fatal error message from the Arduino IDE:
...
-* Change the USB port.
+* Check your USB cable and try a new one (some cables are only for charging and there is no data connection).
+* Change the USB port - prefer direct connection to the computer and avoid USB hubs. Some USB ports may share the power source with other ports used, for example, for charging a phone.
 * Check your power supply.
+* Make sure that nothing is connected to pins labeled **TX** and **RX**.
 * In some instances, you must keep **GPIO0** LOW during the uploading process via the serial interface.
-* Hold down the **“BOOT”** button in your ESP32 board while uploading/flashing.
+* Hold down the **“BOOT”** button on your ESP32 board while uploading/flashing.
+* Solder a **10uF** capacitor in parallel with **RST** and **GND**.
+* If you bought your dev board from a questionable seller, they might have scammed you and sent ESP8266 instead of ESP32.
```
[the remark this answers is not in the document]
Comment (pedrominatel) [reply docs/source/troubleshooting.rst]: I mean the sentence about questionable sellers. This must be removed.

**Review by pedrominatel — APPROVED** [review]
LGTM

**Comment by VojtechBartoska** [comment]
can be merged @me-no-dev

File: libraries/SD/README.md [hunk 4]
```diff
@@ -1,42 +1,52 @@
-
 # SD library
 
-This library provides the integration of ESP32 and SD (Secure Digital) cards without additional modules.
-
+This library provides the integration of ESP32 and SD (Secure Digital) and MMC (Multi Media Card) cards without additional modules. This library is using SPI to interface with the cards. Please not that SPI mode is slower than the intended SD or MMC mode, however, provides more flexibility as the SPI module is available on all ESP SoCs and can be routed to any GPIO through GPIO matrix.
```
Comment (P-R-O-C-H-Y) [inline libraries/SD/README.md:3]: ..._Please_ **note** _that SPI mode_...
(missing e in note)

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
Just 2 small typos, else LGTM!

**Comment by P-R-O-C-H-Y** [comment]
@PilnyTomas Can you please fix those 2 typos so this can be merged? Thanks :)

**Review by P-R-O-C-H-Y — APPROVED** [review]
LGTM

**Pull request merged**